Bumpmap Cleanups

Hoikas opened the pull request:
This changeset fixes the traceback introduced by PyHSPlasma changes by moving bumpmap UV insertion into the main loop. I further took the liberty to make the code more friendly to The Zen of Korman.

Hoikas committed:
Don't preempt UVChannel sanity checks in bumpmaps

This makes UV channel calculations and errors a little bit more sane.

Hoikas committed:
Update some bumpmap variables to obey The Zen

dpogue commented on korman/exporter/mesh.py:
```diff
@@ -267,22 +288,17 @@ def _export_geometry(self, bo, mesh, materials, geospans):
+                    uvs = list([hsVector3(uv[0], 1.0 - uv[1], 0.0) for uv in uvws])
+                    if bumpmap is not None:
+                        uvs.append(dPosDu)
+                        uvs.append(dPosDv)
```
This is slightly different than the original code: In the original, we loop through and each vertex can have multiple UV gradients that are summed and then normalized. In this code, you're just appending already-normalized UV gradients to a list.

Hoikas replied:
oh! thanks, that makes sense now. I still don't like the idea of having to convert the UVs from a tuple to a list over and over and over again, but so be it, I guess...

Hoikas committed:
Update GeoVertex UV touching for HSPlasma changes

plGeometrySpan.TempVertex.uvs now returns an immutable tuple, so we have
to convert that to a list. Furthermore, I took the liberty of changing
to code to avoid spin-washing on the first encounter of a bumpmapped
vertex.

Hoikas commented:
Updated to fix issues from review

dpogue reviewed (approved):
:+1:
